src/menu/menu.c

```diff
@@ -114,23 +114,32 @@ menu_update_width(struct menu *menu)
 			wlr_scene_rect_from_node(item->normal.background),
 			menu->size.width, item->height);
 
-		if (!item->selected.background) {
-			/* This is a separator. They don't have a selected background. */
+		/*
+		 * Separator lines are special because they change width with
+		 * the menu.
+		 */
+		if (item->type == LAB_MENU_SEPARATOR_LINE) {
+			int width = menu->size.width
+				- 2 * theme->menu_separator_padding_width
+				- 2 * theme->menu_item_padding_x;
 			wlr_scene_rect_set_size(
 				wlr_scene_rect_from_node(item->normal.text),
-				menu->size.width - 2 * theme->menu_separator_padding_width,
-				theme->menu_separator_line_thickness);
-		} else {
-			/* Usual menu item */
-			wlr_scene_rect_set_size(
-				wlr_scene_rect_from_node(item->selected.background),
-				menu->size.width, item->height);
-			if (item->native_width > max_width || item->submenu || item->execute) {
-				scaled_font_buffer_set_max_width(item->normal.buffer,
-					max_width);
-				scaled_font_buffer_set_max_width(item->selected.buffer,
-					max_width);
-			}
+				width, theme->menu_separator_line_thickness);
+		}
+
+		if (!item->selected.background) {
+			continue;
+		}
```

Comment:
What exactly does this mean? Does the earlier comment still apply here that this is another way for defining a separator?

Reply:
Will make it more readable (maybe by using `menuitem->type` or something).

Basically, `item->selected` doesn't exist for separator because they're not selectable :smile: ... but that should be a lot more obvious to the reader.

We might be able to get rid of `menu_update_width()` all together, but will try to take small steps.

Reply:
needed for both separator and pipemenu

Reply:
Have just replaced this with `item->selectable` which makes it more readable.